spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/MetricsController.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.dataflow.server.controller;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.concurrent.ExecutionException;
+
+import org.springframework.cloud.dataflow.rest.resource.AppMetricResource;
+import org.springframework.cloud.dataflow.rest.resource.AppMetricResource.MetricsHolder;
+import org.springframework.cloud.dataflow.server.controller.support.ApplicationsMetrics;
+import org.springframework.cloud.dataflow.server.controller.support.ApplicationsMetrics.Application;
+import org.springframework.cloud.dataflow.server.controller.support.ApplicationsMetrics.Instance;
+import org.springframework.cloud.dataflow.server.controller.support.MetricStore;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.data.web.PagedResourcesAssembler;
+import org.springframework.hateoas.ExposesResourceFor;
+import org.springframework.hateoas.PagedResources;
+import org.springframework.hateoas.ResourceAssembler;
+import org.springframework.hateoas.mvc.ResourceAssemblerSupport;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RestController;
+
+/**
+ * Controller handling metrics requests for 'runtime apps'.
+ *
+ * @author Janne Valkealahti
+ *
+ */
+@RestController
+@RequestMapping("/metrics/runtime")
+@ExposesResourceFor(AppMetricResource.class)
+public class MetricsController {
+
+	private final ResourceAssembler<MetricsHolder, AppMetricResource> statusAssembler = new Assembler();
+	private final MetricStore metricStore;
+
+	public MetricsController(MetricStore metricStore) {
+		this.metricStore = metricStore;
+	}
+
+	@RequestMapping(method = RequestMethod.POST)
```

Comment:
While sending json as request body with GET in spring mvc works, angular seem to set it to null. There seem to be other discussions like one I linked that you should not have content body whith GET while according to HTTP specs it is not forbidden. json as request parameter would be a bit brittle.